yt_dlp/extractor/vidyard.py

```diff
@@ -0,0 +1,156 @@
+from .common import InfoExtractor
+from ..utils import int_or_none, traverse_obj
+
+
+class VidyardBaseInfoExtractor(InfoExtractor):
+
+    def _get_formats_and_subtitle(self, video_source, video_id):
+        video_source = video_source or {}
+        formats, subtitles = [], {}
+        for key, value in video_source.items():
+            if key == 'hls':
+                for video_hls in value:
+                    fmts, subs = self._extract_m3u8_formats_and_subtitles(video_hls.get('url'), video_id, headers={
+                        'referer': 'https://play.vidyard.com/'
+                    })
+                    formats.extend(fmts)
+                    self._merge_subtitles(subs, target=subtitles)
+            else:
+                formats.extend({
+                    'url': video_mp4.get('url'),
+                    'ext': 'mp4'
+                } for video_mp4 in value)
+
+        return formats, subtitles
+
+    def _get_direct_subtitle(self, caption_json):
+        subs = {}
+        for caption in caption_json:
+            subs.setdefault(caption.get('language') or 'und', []).append({
+                'url': caption.get('vttUrl'),
+                'name': caption.get('name')
+            })
+
+        return subs
+
+    def _webpage_url(self, url, video_id):
+        return url
+
+    def _real_extract(self, url):
+        video_id = self._match_valid_url(url).group('id')
+        webpage = self._download_webpage(self._webpage_url(url, video_id), video_id)
+
+        json_data = self._download_json(
+            f'https://play.vidyard.com/player/{video_id}.json', video_id)['payload']['chapters'][0]
+
+        formats, subtitles = self._get_formats_and_subtitle(json_data['sources'], video_id)
+        self._merge_subtitles(self._get_direct_subtitle(json_data.get('captions')), target=subtitles)
+
+        return {
```

Comment:
When I looked at this earlier this year I could also get values under these keys (probably not in all cases):
`['chapterId']`
`['milliseconds']` -> `duration` with scale=1000
`['seconds']` -> `duration`
`['thumbnail']['urlSmall']` -> `thumbnail' (may already be included, though)
`['ownerId']` -> `uploader_id`
`['ownerTeamId']` -> `channel_id`